src/Adapter/Product/Image/Uploader/ProductImageUploader.php

```diff
@@ -185,4 +197,37 @@ private function deleteCachedImages(Image $image): void
             }
         }
     }
+
+    /**
+     * Note: we can't delete the whole folder here, or Image::delete will return an error
+     * because it expects the original image and the folder to be present in order to remove
+     * them correctly. So for now this service only handles removing generated image types.
+     * When Image ObjectModel is no longer used, it could also remove the remaining files.
+     *
+     * @param string $imagePath
+     * @param array $imageTypes
+     *
+     * @throws CannotUnlinkImageException
+     */
+    private function deleteGeneratedImages(string $imagePath, array $imageTypes): void
+    {
+        $fileExtension = pathinfo($imagePath, PATHINFO_EXTENSION);
+        $destinationExtension = '.jpg';
+
+        foreach ($imageTypes as $imageType) {
+            $generatedImagePath = sprintf('%s-%s%s', rtrim($imagePath, '.' . $fileExtension), stripslashes($imageType->name), $destinationExtension);
+            if (!file_exists($generatedImagePath)) {
+                continue;
+            }
+
+            if (!unlink($generatedImagePath)) {
```

Comment:
```suggestion
            if (!@unlink($generatedImagePath)) {
```
Because it can raise an e-warning for nothing, and you're already checking the result with CannotUnlinkImageException

Reply:
I think we've decided some time ago to catch the e-warning and pass it right to exception so we know the reason related to filesystem. @matks can you confirm that, I remember we discussed it?
I also have implemented it in Attachments https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/File/Uploader/AttachmentFileUploader.php#L99

Reply:
Adding an @ will fill your error logs, whereas adding a try catch as you did will not :thinking:

Reply:
Raised reminder issue for file | image related concepts
https://github.com/PrestaShop/PrestaShop/issues/23658